app/Services/ValidationService/ValidationRules.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Services;
+
+class ValidationRules
+{
+    const RULES = [
+        'post' => [
+            'author_id' => [
+                'required' => true,
+                'integer'  => true,
+                'exists'   => 'users,id'
+            ],
+            'slug'      => [
+                'required' => true,
+                'string'   => true,
+                'unique'   => 'posts',
+                'regex'    => '/^[a-zA-Z0-9_-]*$/'
```

Comment:
To simplify this expression use case-insensitive regex flag `i` and character class `\d` for digits
so it will be something like that `/^[a-z\d_-]*$/i`

Save or print out this cheatsheet - it will help you many times.
https://i.pinimg.com/originals/5f/91/a2/5f91a29d8276e29e058df23e4d9c51da.png